source/core/TypeAndDataPathFinder.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+
+Copyright (c) 2018 National Instruments Corp.
+
+This software is subject to the terms described in the LICENSE.TXT file
+
+SDG
+*/
+
+/*! \file
+ */
+
+#include "TypeAndDataManager.h"
+#include "DataReflectionVisitor.h"
+
+namespace Vireo
+{
+
+//------------------------------------------------------------
+/* Inherits searching behavior from DataReflectionVisitor to build
+* a path for the specified needle and haystack. Path format does
+* not include the VI name nor "Locals" section.
+*/
+
+class PathFinderVisitor : public DataReflectionVisitor
```

Comment:
Correct me if I am wrong. DataReflectionVisitor does not seem to be used or exported to JS layer except for one test (.via file). In that case, why don't we fix that directly to what we want and update the test result file with the new output? If we don't want to do that, I am curious what use cases due to which we need to maintain DataReflectionVisitor and PathFinderVisitor?